ocaml/lambda/matching.ml

```diff
@@ -100,25 +99,25 @@ open Printpat
 module Scoped_location = Debuginfo.Scoped_location
 
 type error =
-    Non_value_layout of Layout.Violation.t
-  | Illegal_record_field of Layout.const
+    Non_value_jkind of Jkind.Violation.t
```

Comment:
I'm surprised to see this error change.  I think the error is really about layouts.  We might add other similar errors in the future for the other parts of `Jkind`, if there are particularly experimental ones, but I don't think this would grow to encompass them.

Reply:
Yes, I agree here. I've reverted this change.